default.nix

```diff
@@ -67,11 +69,26 @@ in rec {
         else
           ""
       ) (builtins.attrNames pkgConfig));
-    in
+      workspaceJSON = builtins.toJSON {
+        private = true;
+        workspaces = [pname] ++ lib.mapAttrsToList (name: x: "deps/${x.pname}") workspaceDependencies;
+      };
+      workspaceDependencyLinks = lib.concatStringsSep "\n" (lib.mapAttrsToList (name: dep:
+        ''
+          mkdir -p deps/${dep.pname}
+          ln -s ${dep.packageJSON} deps/${dep.pname}/package.json
+        '') workspaceDependencies);
+      workspaceDependencyRemoves =
+        "rm -f ${lib.concatMapStringsSep
+          " "
+          (x: "node_modules/${x}")
+          ([pname] ++ (lib.mapAttrsToList (name: x: x.pname) workspaceDependencies))}";
+in
     stdenv.mkDerivation {
-      inherit name preBuild;
+      inherit name preBuild workspaceJSON;
       phases = ["configurePhase" "buildPhase"];
       buildInputs = [ yarn nodejs ] ++ extraBuildInputs;
+      passAsFile = [ "workspaceJSON" ];
```

Comment:
This is an obscure feature of nix. Do you mind using `pkgs.writeText` to create a derivation out of `workspaceJSON` and pass this instead?

Reply:
Sure.